packages/@aws-cdk/aws-sns/lib/topic.ts

```diff
@@ -70,6 +84,8 @@ export class Topic extends TopicBase {
       displayName: props.displayName,
       topicName: this.physicalName,
       kmsMasterKeyId: props.masterKey && props.masterKey.keyArn,
+      contentBasedDeduplication: props.contentBasedDeduplication,
```

Comment:
So reading the docs it seems `contentBaseddeduplication` can only be true when `fifo` is `true`. That being the case in L2s we prefer to check for this during synth. IE: if a user passed `contentBasedDeduplication: true` without setting `fifo: true`, we should throw an error with a message like "`contentDeduplication` can only be enabled for fifo topics."

Reply:
Thanks for this feedback @MrArnoldPalmer, I'll amend this later today.